HopsanCore/src/Parameters.cpp

```diff
@@ -710,17 +710,22 @@ bool ParameterEvaluatorHandler::evaluateParameter(const HString &rName, HString
 {
     bool success = false;
     // Try our own parameters
-    for(size_t i = 0; i < mParameters.size(); ++i)
+    if(rName.startsWith("self#"))
     {
-        if ( (mParameters[i]->getName() == rName) && (mParameters[i]->getType() == rType) )
+        HString name =rName.substr(5,rName.size()-5);
+        for(size_t i = 0; i < mParameters.size(); ++i)
         {
-            success = mParameters[i]->evaluate(rEvaluatedParameterValue);
+            if ( (mParameters[i]->getName() == name) && (mParameters[i]->getType() == rType) )
+            {
+                success = mParameters[i]->evaluate(rEvaluatedParameterValue);
+            }
         }
     }
-    if(!success)
+    else
     {
         // Try one of our system parent component parameters
-        success = evaluateInSystemParent(rName, rEvaluatedParameterValue, rType);
+        HString name = "self#"+rName;   //Need to add self# for lookup in parent system
```

Comment:
I think that this break the previous behavior. You could then recurse upwards in the model hierarchy (I think) until the desired value was found.

Now only the immediate parent will be visited. 

Also I think that it is kind of ugly to to this by force perpending "self", I would prefer class member function evaluateOwnParameter (or something similar). Need to think about this....

Reply:
That's true, perhaps the "self." should be parsed at a higher level in the evaluateParameter hierarchy (which is a very long chain of functions right now). Then we can just add an ignoreSystemParameters argument which is false by default and set to true when "self." is encountered.

Reply:
Yes the function calling here is a nightmare.